src/libraries/System.IO.FileSystem/tests/Base/BaseGetSetTimes.cs

```diff
@@ -119,25 +132,51 @@ public void SettingUpdatesPropertiesOnSymlink(bool targetExists)
             T link = CreateSymlinkToItem(target);
             if (!targetExists)
             {
-                SettingUpdatesPropertiesCore(link);
+                if (!SettingPropertiesUpdatesLink)
+                {
+                    SettingUpdatesPropertiesCore(target, link);
+                }
             }
             else
             {
                 // Get the target's initial times
                 IEnumerable<TimeFunction> timeFunctions = TimeFunctions(requiresRoundtripping: true);
-                DateTime[] initialTimes = timeFunctions.Select((funcs) => funcs.Getter(target)).ToArray();
+                DateTime[] initialTimes = timeFunctions
+                    .Select((func) => func.Getter(target))
+                    .ToArray();
```

Comment:
```
                // Get initial times that shouldn't changed by SettingUpdatesPropertiesCore.
                IEnumerable<TimeFunction> timeFunctions = TimeFunctions(requiresRoundtripping: true);
                T getTarget = SettingPropertiesUpdatesLink ? target : link;
                DateTime[] initialTimes = timeFunctions.Select((funcs) => funcs.Getter(getTarget)).ToArray();
```

Reply:
Here we want to verify something doesn't change. We store what shouldn't change in `getTarget`.